File: app_tv/src/main/java/tpo/mediaplayer/app_tv/DeviceDao.java
package tpo.mediaplayer.app_tv;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface DeviceDao {
    @Query("SELECT * FROM device")
    List<Device> getAll();

    @Insert
    void insert(Device device);

    @Delete
    void delete(Device device);
}

File: app_phone/src/main/java/tpo/mediaplayer/app_phone/db/DeviceDao.kt
package tpo.mediaplayer.app_phone.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface DeviceDao {
    @Query("SELECT * FROM Device WHERE uid = :uid")
    fun getDeviceByUid(uid: Int): Device?

    @Query("SELECT * FROM Device")
    fun getAllDevices(): List<Device>

    @Insert
    fun insertDevice(device: Device)

    @Query("DELETE FROM Device WHERE uid = :uid")
    fun deleteDeviceByUid(uid: Int)
}